### DISCUSSION TITLE
blog/angular-core-module-standalone-migration

### DISCUSSION THREAD

**Discussion by giscus** [angular-experts-io/angular-experts-website-comments#15] [General]
# blog/angular-core-module-standalone-migration

Let's learn how to migrate commonly used Angular CoreModule (or any other Angular module) to standalone APIs to fully embrace standalone project setup!

https://angularexperts.io/blog/angular-core-module-standalone-migration

**Comment by cristianqr** [comment]
Great post Tomas!

**Comment by PhilippeMalric** [comment]
Beautiful!

**Comment by Arthur-Fedotiev** [comment]
Thank you for 'as always great' content!)
You've mentioned that CORE_GUARD approach could be implemented differently and I indeed did it differently migrating from modules. I just wonder if you can see any potential drawbacks of using simple function and flag which says if dependency was provided instead of DI approach mentioned in your article
```
export const ensureProvidedOnceFactory = (name = 'Dependencies') => {
  let isProvided = false;

  return () => {
    if (isProvided) {
      throw new Error(`${name} can only be provided once.`);
    }

    isProvided = true;
  };
};

const coreGuard = ensureProvidedOnceFactory('Core Dependencies');

export const provideCore = () => {
  coreGuard()
  
 /// other code
```

**Reply by tomastrajan** [reply]
Interesting, and makes perfect sense, will have to explore if I can find some edge cases when this wont be enough but I can't think of any from top of my head 👍
